langchain4j/src/main/java/dev/langchain4j/store/embedding/chroma/ChromaEmbeddingStore.java

```diff
@@ -0,0 +1,172 @@
+                .map(ChromaEmbeddingStore::generateRandomId)
+                .collect(toList());
+
+        addAllInternal(ids, embeddings, textSegments);
+
+        return ids;
+    }
+
+    private void addInternal(String id, Embedding embedding, TextSegment textSegment) {
+        addAllInternal(singletonList(id), singletonList(embedding), textSegment == null ? null : singletonList(textSegment));
+    }
+
+    private void addAllInternal(List<String> ids, List<Embedding> embeddings, List<TextSegment> textSegments) {
+        EmbeddingsRequest embeddingsRequest = EmbeddingsRequest.builder()
+                .embeddings(embeddings.stream()
+                        .map(Embedding::vector)
+                        .collect(toList()))
+                .ids(ids)
+                .metadatas(textSegments == null
+                        ? null
+                        : textSegments.stream()
+                        .map(TextSegment::metadata)
+                        .map(Metadata::asMap)
+                        .collect(toList()))
+                .documents(textSegments == null
+                        ? null
+                        : textSegments.stream()
+                        .map(TextSegment::text)
+                        .collect(toList()))
+                .build();
+
+        chromaClient.addEmbedding(collectionId, embeddingsRequest);
+    }
+
+    @Override
+    public List<EmbeddingMatch<TextSegment>> findRelevant(Embedding referenceEmbedding, int maxResults) {
+        return findRelevant(referenceEmbedding, maxResults, 0);
+    }
+
+    @Override
+    public List<EmbeddingMatch<TextSegment>> findRelevant(Embedding referenceEmbedding, int maxResults, double minScore) {
+        float[] vector = referenceEmbedding.vector();
+        QueryRequest queryRequest = QueryRequest.builder()
+                .queryEmbedding(singletonList(IntStream.range(0, vector.length)
+                        .mapToObj(i -> vector[i])
+                        .collect(toList())))
+                .nResults(maxResults)
+                .build();
+
+        SuccessfulResponse nearestNeighbors = chromaClient.getNearestNeighbors(collectionId, queryRequest);
+
+        return toEmbeddingMatches(nearestNeighbors);
+    }
+
+    private static List<EmbeddingMatch<TextSegment>> toEmbeddingMatches(SuccessfulResponse nearestNeighbors) {
+        List<EmbeddingMatch<TextSegment>> embeddingMatches = new ArrayList<>();
+        for (int i = 0; i < nearestNeighbors.getIds().get(0).size(); i++) {
+            EmbeddingMatch<TextSegment> textSegmentEmbeddingMatch = new EmbeddingMatch<>(
+                    distanceToScore(nearestNeighbors.getDistances().get(0).get(i)),
+                    nearestNeighbors.getIds().get(0).get(i),
+                    Embedding.from(nearestNeighbors.getEmbeddings().get(0).get(i)),
+                    nearestNeighbors.getDocuments().get(0).get(i) == null ? null : TextSegment.from(nearestNeighbors.getDocuments().get(0).get(i),
```

Comment:
please refactor, it is hard to understand

Reply:
changed, but I'm not sure if it's better now